flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/RestHandlerConfiguration.java

```diff
@@ -96,13 +105,20 @@ public static RestHandlerConfiguration fromConfiguration(Configuration configura
 
         final boolean webSubmitEnabled = configuration.getBoolean(WebOptions.SUBMIT_ENABLE);
         final boolean webCancelEnabled = configuration.getBoolean(WebOptions.CANCEL_ENABLE);
+        final boolean webRescaleSupported =
+                JobManagerOptions.SchedulerType.Adaptive.equals(
```

Comment:
Please use ClusterOptions#isAdaptiveSchedulerEnabled and ClusterOptions#isReactiveModeEnabled.